File: scenarios/resource-manager-java-resource-group-crud/golden/src/main/java/com/example/ResourceGroupCrud.java
package com.example;

import com.azure.core.exception.HttpResponseException;
import com.azure.core.management.AzureEnvironment;
import com.azure.core.management.exception.ManagementException;
import com.azure.core.management.profile.AzureProfile;
import com.azure.identity.DefaultAzureCredential;
import com.azure.identity.DefaultAzureCredentialBuilder;
import com.azure.resourcemanager.AzureResourceManager;
import com.azure.resourcemanager.resources.models.ResourceGroup;

public final class ResourceGroupCrud {
    private ResourceGroupCrud() {
    }

    public static void main(String[] args) {
        String subscriptionId = requireEnvironment("AZURE_SUBSCRIPTION_ID");
        String resourceGroupName = requireEnvironment("RESOURCE_GROUP_NAME");

        DefaultAzureCredential credential =
                new DefaultAzureCredentialBuilder().build();
        AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
        AzureResourceManager azure = AzureResourceManager
                .authenticate(credential, profile)
                .withSubscription(subscriptionId);

        try {
            ResourceGroup created = azure.resourceGroups()
                    .define(resourceGroupName)
                    .withRegion("eastus")
                    .create();
            System.out.printf(
                    "Created resource group %s in %s%n",
                    created.name(),
                    created.regionName());

            for (ResourceGroup group : azure.resourceGroups().list()) {
                System.out.println(
                        "Resource group: " + group.name() + " " + group.id());
            }

            ResourceGroup retrieved =
                    azure.resourceGroups().getByName(resourceGroupName);
            System.out.println("Retrieved resource group: " + retrieved.id());

            ResourceGroup updated = retrieved.update()
                    .withTag("environment", "development")
                    .apply();
            System.out.println("Updated resource group tags: " + updated.tags());

            azure.resourceGroups().deleteByName(resourceGroupName);
            System.out.println("Deleted resource group: " + resourceGroupName);
        } catch (ManagementException exception) {
            System.err.println(
                    "Resource management request failed: "
                            + exception.getMessage());
            throw exception;
        } catch (HttpResponseException exception) {
            System.err.println(
                    "Azure request failed: " + exception.getMessage());
            throw exception;
        }
    }

    private static String requireEnvironment(String name) {
        String value = System.getenv(name);
        if (value == null || value.isBlank()) {
            throw new IllegalStateException(
                    "Set " + name + " before running.");
        }
        return value;
    }
}

File: scenarios/resource-manager-python-resource-group-crud/golden/resource_group_crud.py
from __future__ import annotations

import os
import sys

from azure.core.exceptions import HttpResponseError
from azure.identity import DefaultAzureCredential
from azure.mgmt.resource import ResourceManagementClient
from azure.mgmt.resource.resources.models import ResourceGroup


def run() -> None:
    subscription_id = os.environ["AZURE_SUBSCRIPTION_ID"]
    resource_group_name = os.environ["AZURE_RESOURCE_GROUP_NAME"]
    location = os.environ.get("AZURE_LOCATION", "eastus")

    credential = DefaultAzureCredential()
    client = ResourceManagementClient(credential, subscription_id)

    try:
        parameters = ResourceGroup(location=location)
        created = client.resource_groups.create_or_update(
            resource_group_name,
            parameters,
        )
        print(f"Created: {created.name} in {created.location}")

        for resource_group in client.resource_groups.list():
            print(f"Resource group: {resource_group.name}")

        retrieved = client.resource_groups.get(resource_group_name)
        print(f"Retrieved: {retrieved.name} in {retrieved.location}")

        updated = client.resource_groups.update(
            resource_group_name,
            {"tags": {"environment": "development"}},
        )
        print(f"Updated tags: {updated.tags}")

        deletion = client.resource_groups.begin_delete(resource_group_name)
        deletion.result()
        print(f"Deleted resource group: {resource_group_name}")
    except HttpResponseError as error:
        print(f"Resource group operation failed: {error}", file=sys.stderr)
        raise


if __name__ == "__main__":
    run()
